test/pkg/environment/monitor.go

```diff
@@ -106,6 +108,12 @@ func (m *Monitor) CreatedNodes() int {
 	return m.TotalNodesSeen() - m.numberNodesAtReset
 }
 
+func (m *Monitor) GetCreatedNodes() []v1.Node {
+	return lo.Filter(m.GetNodes(), func(node v1.Node, _ int) bool {
+		return node.Labels[v1alpha5.ProvisionerNameLabelKey] != ""
```

Comment:
With some parallelism, this can be wrong, but for running tests in serial, this is fine.

Reply:
The previous check would also be wrong in parallel.